src/shared/common/GpiiStore.js

```diff
@@ -219,22 +176,42 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
 
     gpii.prefs.gpiiStore.set = function (model, settings, session, modelTransformFunc, onSuccessfulSetFunction) {
         var transformedModel = modelTransformFunc(model);
+        
+        var dataToPost = {
+            "contexts": {
+                "gpii-default": {
+                    "name": "Default preferences",
+                    "preferences": transformedModel
+                }
+            }
+        };
+        var urlToPost, requestType;
+        if (session.options.loggedUser) {
+            urlToPost = settings.url + "preferences/" + session.options.loggedUser;
```

Comment:
The goal of this pull request is to make the PCP work with the API changes that resulted from the Fireball work. Is the issue with the URL construction serious enough to block this pull request? Can't this be handled in a separate JIRA ticket?